File: Array/1909. Remove One Element to Make the Array Strictly Increasing/canBeIncreasing.go
func canBeIncreasing(nums []int) bool {
	for i := 0; i < len(nums); i++ {
		temp := make([]int, 0, len(nums)-1)
		for j := 0; j < len(nums); j++ {
			if i != j {
				temp = append(temp, nums[j])
			}
		}
		increasing := true
		for k := 0; k < len(temp)-1; k++ {
			if temp[k] >= temp[k+1] {
				increasing = false
				break
			}
		}
		if increasing {
			return true
		}
	}
	return false
}

File: Array/1909. Remove One Element to Make the Array Strictly Increasing/CanBeIncreasing.cpp
#include <vector>
#include <iostream>

class Solution {
public:
    bool canBeIncreasing(vector<int>& nums) {
        for(int i=0;i<nums.size();i++){
            std::vector<int> myvector;
            for(int j=0;j<nums.size();j++){
                if(i!=j){
                    myvector.push_back(nums[j]);
                }
            }
            bool increasing=true;
            for(int k=0;k<myvector.size()-1;k++){
                if(myvector[k]>=myvector[k+1]){
                    increasing=false;
                }
            }
            if(increasing==true){
                return true;
            }
        }
        return false;
    }
};

// Example usage and test function
int main() {
    Solution sol;
    std::vector<int> nums1 = {1, 2, 10, 5, 7};
    std::vector<int> nums2 = {2, 3, 1, 2};
    
    std::cout << "Test 1: " << (sol.canBeIncreasing(nums1) ? "Pass" : "Fail") << std::endl;
    std::cout << "Test 2: " << (sol.canBeIncreasing(nums2) ? "Pass" : "Fail") << std::endl;

    return 0;
}